### ISSUE TITLE
Allow durations other than `day` for `seq date`

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#13816]
### Related problem

From [Discord today](https://discord.com/channels/601130461678272522/1282666419590533161), currently `seq date` can only use `--date <int>`, which isn't very "Nu'ish".

### Describe the solution you'd like

Allow `seq date --duration <duration>` to generate `datetime` values with "step" values based on arbitrary durations, rather than just days.

### Describe alternatives you've considered

(from @fdncred) `seq datetime` as an alternative command

(from @weirdan) Or a range syntax that accepts a second value in the stride.

**Comment by NotTheDr01ds** [comment]
Side-note:  `seq date` (with or without this enhancement) is probably a good candidate to remove from core and move to `std-lib` when startup performance issues with `std-lib` are resolved.  Given that, I'd personally prefer that `std-lib` be prioritized higher than this change ;-)

**Comment by fdncred** [comment]
The problem with our range is the second number in the stride isn't the "count by" number. However, it is in seq.
Example:
range sequences are count by mid - first until last
```nushell
❯ 1..3..12
╭───┬────╮
│ 0 │  1 │
│ 1 │  3 │
│ 2 │  5 │
│ 3 │  7 │
│ 4 │  9 │
│ 5 │ 11 │
╰───┴────╯
```
seq is first to last count by mid
```nushell
❯ seq 1 3 12
╭───┬────╮
│ 0 │  1 │
│ 1 │  4 │
│ 2 │  7 │
│ 3 │ 10 │
╰───┴────╯
```

**Comment by NotTheDr01ds** [comment]
Agreed that the mismatch between `seq` and range-literal syntax is an issue, but the `seq date` mismatch is completely different since it uses a completely different syntax.  Definitely consistency issues all around here.

I don't know how much of a problem this is in reality, because it's always possible to convert from one to the other.

I think the bigger issue I have with dates-in-ranges is that it's just not very readable:

```
2024-02-01T00:00:00..2024-02-01T00:02:00..2024-02-01T10:00:00
```

Any `seq date` syntax is an improvement, IMHO.
